.changeset/store-core-peer-floors.md

```
@@ -0,0 +1,21 @@
+'@mastra/dsql': patch
+'@mastra/dynamodb': patch
+'@mastra/lance': patch
+'@mastra/mongodb': patch
+'@mastra/pg': patch
+'@mastra/redis': patch
+'@mastra/upstash': patch
+---
+
+Fixed nine storage adapters declaring a `@mastra/core` peer range that permitted core versions too old to load them. Each adapter imports `storageMessageMatchesMetadataFilter` from `@mastra/core/storage`, which core only exports from 1.53.0, but every one of them still advertised a floor below that — as low as `>=1.0.0-0`. Package managers accepted the incompatible pair without a warning and the install then failed at import time:
+
+```
```

markdownlint-cli2 (0.23.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)